lib/Twig/Extension/Core.php

```diff
@@ -1272,7 +1276,11 @@ function twig_length_filter(Twig_Environment $env, $thing)
             return mb_strlen((string) $thing, $env->getCharset());
         }
 
-        return count($thing);
+        if ($thing instanceof \Countable || is_array($thing)) {
+            return count($thing);
+        }
+
+        return 1;
```

Comment:
Wouldn't be better to throw an exception and not return a magic `1`?
Something like;
```php
throw new Twig_Error_Runtime(sprintf('Cannot determine length of "%s", expect scalar, array or Countable', get_class($thing)));
```